codegen/cmd/injection-gen/generators/reconciler_reconciler.go

```diff
@@ -320,6 +328,24 @@ func (r *reconcilerImpl) Reconcile(ctx {{.contextContext|raw}}, key string) erro
 		}
 	}
 
+	{{if .genDuckLogic}}
+	// Bump observed generation to denote that we have processed this
+	// generation regardless of success or failure.
+	resource.Status.ObservedGeneration = resource.Generation
+
+	if resource.Status.ObservedGeneration != original.Status.ObservedGeneration && reconcileEvent != nil {
+		originalRc := original.Status.GetCondition(apis.ConditionReady)
+		rc := resource.Status.GetCondition(apis.ConditionReady)
+		// if a new generation is observed and reconciliation reported an error event
+		// the reconciler should change the ready state. By default we will set unknown.
+		if equality.Semantic.DeepEqual(originalRc, rc) {
```

Comment:
but this could be that it went from ready true to ready true if the update was something minor

Reply:
This only runs if reconcileEvent != nil should we should be in a (potential) error case.